### PR TITLE
if getJson fails also use cached sources on updating repository

### LINKED ISSUE #963: Failed repo updates should not delete the currently known repo
https://forum.iobroker.net/topic/34891/update-fehler-und-adapter-liste-leer

### PR DESCRIPTION
**Pull request by foxriver76**
- should fix #963

### CODE REVIEW

File: lib/tools.js [hunk 1]
```diff
@@ -998,20 +998,20 @@ function getRepositoryFile(urlOrPath, additionalInfo, callback) {
...
                             _getRepositoryFile(sources, path, err => {
-                                err && console.error('[' + new Date() + '] ' + err);
+                                err && console.error(`[${new Date()}] ${err}`);
                                 typeof callback === 'function' && callback(err, sources, actualSourcesHash);
                             }));
                     } else {
-                        if (typeof callback === 'function') {
-                            callback('Cannot read "' + urlOrPath + '"', {});
-                        }
+                        // return cached sources, because no sources found
```
Comment (foxriver76) [inline lib/tools.js:1012]: relevant part, maybe we should also return no error like the other methods do, which return cached sources

Comment (Apollon77) [reply lib/tools.js:1012]: The problem is that logging to console is invisible mainly ... so the error is then logged in the calling methid I think

**Comment by Apollon77** [comment]
LGTM

File: lib/tools.js (hunk 1 above)
Comment (Apollon77) [reply lib/tools.js:1012]: Or other ... we would need to check how the "calling code" works ... if they nly see "err" and then cancel execution we should not return error

Comment (Apollon77) [reply lib/tools.js:1012]: Ok should work with returning an error. all "external" users of getRepositoryFile method I found logged error when it was set, but then check for sources existence

**Comment by AlCalzone** [comment]
If all calling methods know that the callback is not compatible with the usual convention, LGTM.

**Pull request merged**